src/network/networkd-routing-policy-rule.c

```diff
@@ -1211,6 +1214,30 @@ int config_parse_routing_policy_rule_prefix(
         if (r < 0)
                 return log_oom();
 
+        if (streq(lvalue, "To")) {
+                if (streq(rvalue, "_dhcp4_ip"))
+                        n->to_dhcp_ip = true;
+                else if (streq(rvalue, "_dhcp4_gw"))
+                        n->to_dhcp_gateway = true;
+
+                if (n->to_dhcp_ip || n->to_dhcp_gateway) {
+                        TAKE_PTR(n);
+                        return 0;
+                }
+        }
+
+        if (streq(lvalue, "From")) {
+                if (streq(rvalue, "_dhcp4_ip"))
+                        n->from_dhcp_ip = true;
+                else if (streq(rvalue, "_dhcp4_gw"))
+                        n->from_dhcp_gateway = true;
+
+                if (n->from_dhcp_ip || n->from_dhcp_gateway) {
+                        TAKE_PTR(n);
+                        return 0;
+                }
+        }
+
```

Comment:
Please check `IN_SET(n->family, AF_UNSPEC, AF_INET)` before setting the flag. That is, refuse the case, e.g.,
```
From=2001::1
To=_dhcp4_ip
```
Also, set `n->family = AF_INET;` when enabling these flags.